go/vt/worker/topo_utils.go

```diff
@@ -119,11 +119,19 @@ func FindWorkerTablet(ctx context.Context, wr *wrangler.Wrangler, cleaner *wrang
 		return nil, err
 	}
 
+	wr.Logger().Infof("Changing tablet %v to '%v'", topoproto.TabletAliasString(tabletAlias), topodatapb.TabletType_DRAINED)
+	shortCtx, cancel := context.WithTimeout(ctx, *remoteActionsTimeout)
+	err = wr.ChangeSlaveType(shortCtx, tabletAlias, topodatapb.TabletType_DRAINED)
+	cancel()
+	if err != nil {
+		return nil, err
+	}
+
```

Comment:
Oh good catch. I think we can solve this by refreshing tablet state after adding this tag. Let's get @sougou  input on this.

Reply:
I looked at implementation of `ChangeSlaveType` and it has a built-in refresh, which is probably why the tag change is done before.

The refresh that @rafael talks about is a higher level one, and I think it's for reacting to change in serving state. But since this is a lower level function that others can use independent of serving state, we should preserve the original order.